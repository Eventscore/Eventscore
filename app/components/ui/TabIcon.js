//<TabIcon icon={'search'} selected={false} />
import React, { PropTypes } from 'react';
import { Icon } from 'react-native-elements';

import { AppColors } from '@theme/';

const TabIcon = ({ icon, selected }) => (
  <Icon
    name={icon}
    size={26}
    color={selected ? AppColors.tabbar.iconSelected : AppColors.tabbar.iconDefault}
  />
);

TabIcon.propTypes = { icon: PropTypes.string.isRequired, selected: PropTypes.bool };
TabIcon.defaultProps = { icon: 'search', selected: false };

export default TabIcon;
