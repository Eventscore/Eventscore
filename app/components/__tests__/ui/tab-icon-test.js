/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import TabIcon from '@ui/TabIcon';

it('TabIcon renders correctly', () => {
  const tree = renderer.create(
    <TabIcon icon={'help'} selected />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
