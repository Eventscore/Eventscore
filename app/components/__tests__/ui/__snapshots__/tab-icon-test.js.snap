// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabIcon renders correctly 1`] = `
<View
  onLongPress={undefined}
  onPress={undefined}
  style={
    Array [
      false,
      false,
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  underlayColor="white"
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#0E4EF8",
          "fontSize": 26,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î¢‡
  </Text>
</View>
`;
